masu/__init__.py

```diff
@@ -55,12 +57,18 @@ def create_app(test_config=None):
         os.makedirs(app.instance_path)
     # pylint: disable=invalid-name
     except OSError as e:
-        logger.warning(e)
+        # ignore "File exists"
+        if e.errno != errno.EEXIST:
+            logger.warning(e)
 
     # Establish database
     # pylint: disable=invalid-name, unused-variable
     db = SQLAlchemy(app)  # noqa: F841
 
+    # Celery task queue
+    proc = Processor()
+    proc.run()
```

Comment:
We might want to consider breaking out the run() from the initializer and have it kicked off by some other means.  That way we can have some more control in the test environment. For example: See if a task is ready (should return false), run a long task, check if task is done during and at the end.

Reply:
Agree, but I'm expecting that to come with additional development beyond this very rudimentary scaffolding. Most of that requires spec'ing out interactions we haven't yet defined.

Reply:
I think we should be using the celery app factory you created in worker.py here. And it really just becomes our plugin point to celery.

```
celery = make_celery(app)
```

Reply:
I think the main issue is what comes next?  We have to call [task].delay() somewhere. So, where does that get called if `get_app` is calling `make_celery` ?

The options I can think of are:

1. call tasks in `get_app`, which is the Big Ball Of Mud design pattern, and is not ideal.
2. create an object and pass in the celery object obtained from `make_celery` into our task-running object, which is acceptable but begs the question of why doesn't that object's constructor just use the factory directly?
3. Insert an object between `get_app` and the celery factory to handle the interface between Flask and our processing tasks. Maybe we absorb the factory method into our object, and maybe we don't. This is where I stopped short, to spend time looking at other parts of the design before committing to pushing the factory into `Processor` entirely.

Reply:
We call [task].delay() wherever/whenever we want to put a task in the queue. In the SNS example, that would be done in the view that handles the endpoint. In the polling case if we switch to using the scheduler, then we don't call `.delay()` anywhere.